pkg/resource/deploy/step_generator.go

```diff
@@ -601,6 +601,16 @@ func (sg *stepGenerator) generateStepsFromDiff(
 	// If there were changes check for a replacement vs. an in-place update.
 	if diff.Changes == plugin.DiffSome {
 		if diff.Replace() {
+			// If this resource is protected we can't replace it because that entails a delete
+			// Note that we do allow unprotecting and replacing to happen in a single update
+			// cycle, we don't look at old.Protect here.
+			if goal.Protect {
+				return nil, result.Errorf("unable to replace resource %q\n"+
+					"as it is currently marked for protection. To unprotect the resource, "+
+					"either remove the `protect` flag from the resource in your Pulumi"+
+					"program and run `pulumi up`", urn)
```

Comment:
If we have `either`, we need to provide >1 options. Is this what you meant?
```suggestion
					"program and run `pulumi up` or edit the sate and remove the protect flag manually", urn)
```

Reply:
Oh bad copy paste from the other protect error message. The only thing that makes sense here is changing the program (because we're look at Goal.Protect)